src/utils/jsonwebtoken.js

```diff
@@ -0,0 +1,124 @@
+// packages
+const jwt = require('jsonwebtoken');
+const dotenv = require('dotenv').config();
+const crypto = require('crypto');
+const fs = require('fs');
+const path = require('path');
+
+/** 
+ * @param {object} userInfo - validated user id
+ * @param {string} expirationTime - token validity
+ */
+
+/**Basic / symmetric jwt generation */
+const generatingToken = function (userInfo, expirationTime) {
+    // sample of generated payload
+    const {userId, fullName, email} = userInfo;
+    const payload = {
+        sub: userId,
+        name: fullName,
+        email,
+        iat: Date.now(),
+    }
+
+    // signing jwt
+    const token = jwt.sign(payload, process.env.SECRET_KEY, {expiresIn: expirationTime})
+    return token;
+}
+
+const refreshToken = function(expirationTime = '1day') {
+    return  generatingToken(userInfo, expirationTime);
+}
+
+const accessToken = function(expirationTime = '10m') {
+    return {
+        access_token: generatingToken(userInfo, expirationTime),
+        token_type: 'Bearer',
+        expires_in: '10m',
+        refresh_token: refreshToken(),
```

Comment:
How do i get user_info into this functions (refreshToken and accessToken), you might want to add user_info as parameters to the functions.

Should the refresh token be inside the accessToken?, the refresh and access token should just return only the token itself, every information that they should carry should be inside the payload being encoded.

I feel the accessToken function should be just like the refreshToken function, the only difference is the expirationTime, and maybe the path for the refresh token but that might not be necessary now though.

Reply:
Yes it should, noted. 
The initial thought was to bundle it together to be easily accessible when access token is called, however this would defeat the purpose of the refresh token since it'll always be generated anew. Thanks